src/audio/mixer/mixer.c

```diff
@@ -40,195 +39,86 @@ DECLARE_SOF_RT_UUID("mixer", mixer_uuid, 0xbc06c037, 0x12aa, 0x417c,
 
 DECLARE_TR_CTX(mixer_tr, SOF_UUID(mixer_uuid), LOG_LEVEL_INFO);
 
-/* mixer component private data */
-struct mixer_data {
-	bool sources_inactive;
-
-	void (*mix_func)(struct comp_dev *dev, struct audio_stream __sparse_cache *sink,
-			 const struct audio_stream __sparse_cache **sources, uint32_t count,
-			 uint32_t frames);
-};
 
-static struct comp_dev *mixer_new(const struct comp_driver *drv,
-				  struct comp_ipc_config *config,
-				  void *spec)
+static int mixer_init(struct processing_module *mod)
 {
-	struct comp_dev *dev;
+	struct module_data *mod_data = &mod->priv;
+	struct comp_dev *dev = mod->dev;
 	struct mixer_data *md;
 
-	comp_cl_dbg(&comp_mixer, "mixer_new()");
-
-	dev = comp_alloc(drv, sizeof(*dev));
-	if (!dev)
-		return NULL;
-	dev->ipc_config = *config;
+	comp_dbg(dev, "mixer_new()");
 
 	md = rzalloc(SOF_MEM_ZONE_RUNTIME, 0, SOF_MEM_CAPS_RAM, sizeof(*md));
-	if (!md) {
-		rfree(dev);
-		return NULL;
-	}
+	if (!md)
+		return -ENOMEM;
 
-	comp_set_drvdata(dev, md);
-	dev->state = COMP_STATE_READY;
-	return dev;
+	mod_data->private = md;
+	mod->verify_params_flags = BUFF_PARAMS_CHANNELS;
+	mod->simple_copy = true;
```

Comment:
doesn't it have to set `.no_pause`?

Reply:
yes, it was added specifically for this :D. Thanks, fixed now.